### PR TITLE
Add whole-body differential IK example

### PR DESCRIPTION
**Pull request by sea-bass**
This PR adds a `joint_weights` option to differential IK, which can be used for e.g. solving whole-body IK.

Incidentally, I also had to fix how the "ignore joints" works, meaning we're now doing it better. Meaning now we grab only the columns of the Jacobian we need instead of doing the full calculation and throwing away unneeded output.

It also adds a UR5 on mobile base URDF and cleans up all the models in the package to be in the same location. (this is where most of the deletions come from)

```
python3 examples/differential_ik_whole_body.py
```

image

Closes https://github.com/sea-bass/pyroboplan/issues/52

### CODE REVIEW

**Comment by sea-bass** [comment]
One internal debate I am having is whether this qualifies as "whole body IK" or I should relax some of the claims and use something like mobile base, floating base, etc.

File: examples/intro_pinocchio/intro_pinocchio_ur5_sim.py [hunk 1]
```diff
@@ -59,7 +60,7 @@ def main():
 
     # Set up initial conditions for the simulation.
     q = np.array([0.0, -np.pi / 2 + 0.01, 0.0, 0.0, 0.0, 0.0])
-    v = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+    v = np.array([1.0, 0.0, 0.0, 0.0, 0.0, 0.0])
```
Comment (sea-bass) [inline examples/intro_pinocchio/intro_pinocchio_ur5_sim.py:63]:
```suggestion
    v = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0])

File: examples/differential_ik_whole_body.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    # Initialize visualizer
+    viz = MeshcatVisualizer(model, collision_model, visual_model, data=data)
+    viz.initViewer(open=True)
+    viz.loadViewerModel()
+    np.set_printoptions(precision=3)
+
+    # Set up the IK solver.
+    # The first 3 values in the weights are the base, and the other 6 are the arm.
+    # This means we are weighing base motion more than arm motion.
```
Comment (eholum) [inline examples/differential_ik_whole_body.py:45]: Just to be clear, higher weight -> increased "cost" for moving that joint -> less overall motion is expected from the joint?

File: src/pyroboplan/ik/differential_ik.py [hunk 3]
```diff
@@ -54,6 +55,9 @@ def __init__(
             max_step_size : float
                 Maximum gradient step size, between 0 and 1, based on ratio of current distance to target to initial distance to target.
                 To use a fixed step size, set both minimum and maximum values to be equal.
+            joint_weights : list[float], optional
+                A list of relative weights for different joints, used in computing the Jacobian pseudoinverse.
```
Comment (eholum) [inline src/pyroboplan/ik/differential_ik.py:59]: Might be worth adding some clarification on high vs low weights and how it impacts the solver. When I first went through this I inverted... this was the only bit that confused me here.

File: src/pyroboplan/ik/differential_ik.py [hunk 4]
```diff
@@ -157,6 +163,26 @@ def solve(
...
+        if self.options.joint_weights is None:
+            # Use identity weights if they are not specified.
+            W = np.eye(num_active_joints)
+        elif len(self.options.joint_weights) != num_active_joints:
+            raise ValueError(
+                f"Joint weights, if specified, must have {num_active_joints} elements."
+            )
+        else:
+            # Invert the weights so that higher weight means less joint motion.
+            W = np.linalg.inv(np.diag(self.options.joint_weights))
```
Comment (eholum) [inline src/pyroboplan/ik/differential_ik.py:184]: Do you want to add any kind of validation here? Like you can't have 0 as a weight or this becomes singular. Also, what does a negative weight mean?

**Review by eholum — COMMENTED** [review]
I think _whole body_ is appropriate given a very quick googling of where the term is used!

Just some minor questions but overall this is pretty nifty. Plus it's so fun to watch the entire robot run off into the distance when the solution diverges.

File: examples/differential_ik_whole_body.py (hunk 2 above)
Comment (sea-bass) [reply examples/differential_ik_whole_body.py:45]: Yep!

File: src/pyroboplan/ik/differential_ik.py (hunk 4 above)
Comment (sea-bass) [reply src/pyroboplan/ik/differential_ik.py:184]: Good idea.

**Review by eholum — APPROVED** [review]
👍👍👍

**Pull request merged**